### PR TITLE
Label/Priority and Filters

### PR DESCRIPTION
**Pull request by Jia-A**
In this pull request, following features related to dot-note app are added : 
- User can added pre-defined tag on notes.
- User can make a new tag in the Label page and the add it in his/her notes.
- User can set a priority on notes.
- User can filter notes according to tags or priority and both.

### CODE REVIEW

File: src/note-components/notepage/note.jsx [hunk 1]
```diff
@@ -2,49 +2,54 @@ import "./note.css";
...
+const { note, priority } = noteState;
+const { filterState, filteredNotes } = useFilter();
 
 
 const noteCreateFuntion = async (e) => {
 e.preventDefault();
 if (token) {
 noteCreate(notes, token, noteDispatch);
-setNote({ title: "", mainContent: "" });
+console.log(notes.backColor)
```
Comment (Tanay-J) [inline src/note-components/notepage/note.jsx:27]: you missed removing this

File: src/note-components/notepage/note.jsx [hunk 2]
```diff
@@ -2,49 +2,54 @@ import "./note.css";
...
-    }
-    console.log("Done");
+try{
+const response = await axios({
+method : "post",
+data : { note : item },
+url : `api/notes/archives/${item._id}`,
+headers : {authorization : token},
+});
+console.log("try passed");
```
Comment (Tanay-J) [inline src/note-components/notepage/note.jsx:44]: here too

File: src/note-components/notepage/note.css [hunk 3]
```diff
@@ -166,6 +166,19 @@ span {
...
 
+/* input[type="color"] {
+	opacity: 5;
+	display: block;
+    width: 20px;
+    height: 20px;
+	border: none;
+    border-radius: 100%;
+    background: none;
+} */
```
Comment (Tanay-J) [inline src/note-components/notepage/note.css:177]: you may remove this comment

File: src/note-components/label/label.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+const Label = () =>{
+const { tag, setTag } = useNote();
+let newTag= "";
+
+const tagCreationFunction = () =>{
+if(newTag !== ""){
+setTag([...tag, newTag])
+newTag="";
+console.log(newTag)
```
Comment (Tanay-J) [inline src/note-components/label/label.jsx:14]: logging statements can be removed

File: src/note-components/label/label.css [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    padding: 10px;
+    cursor: pointer; 
+}
+.marginned{
+    margin: 0.9375rem;
+    padding: 0px 0px 5px 15px;
+    background-color: rgb(231, 216, 231);
+    border-radius: 5px;
+}
+.margin{
```
Comment (Tanay-J) [inline src/note-components/label/label.css:30]: try to use self explanatory class names

**Review by Tanay-J — APPROVED** [review]
Good work at using context and reducers. You may check the  suggestions I have made. 
Also, 
- Please use a code formatter 
- Please include preview link or video of the functionality for better code review

File: src/note-components/label/label.css [hunk 6]
```diff
@@ -0,0 +1,75 @@
+.right-cont{
+    justify-content: left;
+}
+.chips-box{
+    display: flex;
+    /* flex-direction: column; */
```
Comment (Megha-Joshi) [inline src/note-components/label/label.css:6]: Remove the commented code.

File: src/note-components/label/label.css (hunk 5 above)
Comment (Megha-Joshi) [reply src/note-components/label/label.css:30]: +1

File: src/note-components/label/label.css [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    margin: 5px;
+    background-color: thistle;
+    border: none;
+    border-radius: 1.25rem;
+}
+.new-item:hover{
+    border-bottom: 2px solid rgb(145, 60, 145);
+}
+.list-item:hover{
+    color: white;
```
Comment (Megha-Joshi) [inline src/note-components/label/label.css:70]: Use CSS variable

File: src/Context/note-context.js [hunk 8]
```diff
@@ -55,18 +56,33 @@ const NoteProvider = ({
   const [noteState, noteDispatch] = useReducer(noteReducer, {
     note: [],
     trash: [],
-    archive: []
+    archive: [],
+    tags: [],
+    priority: ["Critical", "Major", "Moderate", "Low"],
+    // filterTagPriority : {Critical : false, Major : false, Moderate : false, Low : false},
```
Comment (Megha-Joshi) [inline src/Context/note-context.js:69]: remove this commented code

**Review by Megha-Joshi — APPROVED** [review]
Everything looks fine to me just remove the commented code and console logs.

**Pull request merged**